### PR TITLE
15 feature settings

### PR DESCRIPTION
**Pull request by jm-go**
- Completed Settings Page
- Added functionality for dark & light mode
- Added modals for Logout buttons
- The functionality for changing fonts is yet to be implemented

### CODE REVIEW

File: src/components/Button/Button.scss [hunk 1]
```diff
@@ -13,13 +13,14 @@
...
     background-color: $clr-accent-grey;
-    color: white;
+    color: $clr-white;
     border: none;
   }
 
   &--red {
     background-color: $clr-accent-red;
     border: none;
+    color: $clr-white;
```
Comment (olirob93) [inline src/components/Button/Button.scss:23]: why is this the same variable as the grey colour above?

File: src/App.tsx [hunk 2]
```diff
@@ -3,16 +3,28 @@ import Home from "./pages/Home/Home";
...
 
 const App = () => {
+  const [theme, setTheme] = useState<"light" | "dark">("light");
+
+  /**
+   * Toggles the app's theme between light and dark modes.
+   * It updates the 'theme' state to either 'light' or 'dark',
+   * depending on its current value. The updated theme is then applied
+   * to the entire app.
+   */
```
Comment (olirob93) [inline src/App.tsx]: no comments here, if it's complex where should it go?

File: src/components/SettingsCard/SettingsCard.tsx [hunk 3]
```diff
@@ -7,13 +7,24 @@ import sun from "../../assets/images/sun.svg";
...
+const SettingsCard = ({
+  variant,
+  onToggleDarkMode,
+  onApplyChanges,
+}: SettingsCardProps) => {
+  const [isToggleOn, setIsToggleOn] = useState<boolean>(variant === "dark");
+
+  /**
+   * Handles the change event when the toggle switch for light/dark mode is clicked.
+   */
```
Comment (olirob93) [inline src/components/SettingsCard/SettingsCard.tsx]: no comments here

File: src/components/Toggle/Toggle.tsx [hunk 4]
```diff
@@ -1,25 +1,30 @@
...
 };
 
-const ToggleSwitch = ({ onToggle } :ToggleSwitchProps) => {
-  const [isChecked, setIsChecked] = useState(false);
-
+const ToggleSwitch = ({ onToggle, checked }: ToggleSwitchProps) => {
+  
+  /**
+   * Handles the toggle event when the toggle switch state changes.
+   */
```
Comment (olirob93) [inline src/components/Toggle/Toggle.tsx]: no comments here

**Review by olirob93 — CHANGES_REQUESTED** [review]
A few things needed here, mainly removing of the comments please into somewhere better or completely.

File: src/components/Button/Button.scss (hunk 1 above)
Comment (jm-go) [reply src/components/Button/Button.scss:23]: image
The font was black not white like on figma (the picture attached), that's why this line was added. I think the settings page is the first page that makes use of the red variant of the button.

**Review by fihockey — APPROVED** [review]

**Review by mh-fung — APPROVED** [review]

**Review by olirob93 — APPROVED** [review]

**Pull request merged**